File: app/src/main/java/com/juniperphoton/myersplash/utils/ToastService.java
package com.juniperphoton.myersplash.utils;

import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.juniperphoton.myersplash.R;
import com.juniperphoton.myersplash.base.App;

public class ToastService {
    public static void sendShortToast(String str) {
        LayoutInflater inflater = LayoutInflater.from(App.getInstance());
        View view = inflater.inflate(R.layout.toast_layout, null);

        TextView textView = (TextView) view.findViewById(R.id.toast_textView);
        textView.setText(str);

        Toast toast = new Toast(App.getInstance());
        toast.setDuration(Toast.LENGTH_SHORT);
        toast.setView(view);
        toast.setGravity(Gravity.BOTTOM, 0, 100);
        toast.show();
    }
}

File: app/src/main/kotlin/com/juniperphoton/myersplash/utils/ToastService.kt
package com.juniperphoton.myersplash.utils

import android.annotation.SuppressLint
import android.os.Handler
import android.os.Looper
import android.view.Gravity
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast
import com.juniperphoton.myersplash.App
import com.juniperphoton.myersplash.R

object ToastService {
    private var handler = Handler(Looper.getMainLooper())

    fun sendShortToast(str: String?) {
        if (str == null) {
            return
        }
        if (Looper.getMainLooper() != Looper.myLooper()) {
            handler.post { sendToastInternal(str) }
        } else {
            sendToastInternal(str)
        }
    }

    fun sendShortToast(strId: Int) {
        if (strId == 0) {
            return
        }
        if (Looper.getMainLooper() != Looper.myLooper()) {
            handler.post { sendToastInternal(App.instance.getString(strId)) }
        } else {
            handler.post { sendToastInternal(App.instance.getString(strId)) }
        }
    }

    @SuppressLint("InflateParams")
    private fun sendToastInternal(str: String?) {
        val view = LayoutInflater.from(App.instance).inflate(R.layout.toast_layout, null)

        val textView = view.findViewById<TextView>(R.id.toast_textView)
        textView.text = str

        val toast = Toast(App.instance)
        toast.duration = Toast.LENGTH_SHORT
        toast.view = view
        toast.setGravity(Gravity.BOTTOM, 0, 100)
        toast.show()
    }
}